src/vulkan-renderer/application.cpp

```diff
@@ -620,6 +611,12 @@ void Application::run() {
         update_imgui_overlay();
         render_frame();
         process_mouse_input();
+        if (m_input_data->was_key_pressed_once(GLFW_KEY_N)) {
+            load_octree_geometry();
+            generate_octree_indices();
+            m_index_buffer->upload_data(m_octree_indices);
+            m_vertex_buffer->upload_data(m_octree_vertices);
+        }
```

Comment:
This will add another random world with seed `42` **not** another random world, also the existing world should be deleted as they will overlap.

Reply:
Right, I kinda rushed rebasing on top of the collision so I didn't notice the overlapping. As for the randomness though, it seems to generate a different world every time so I'm not sure if there's any problem there?

Reply:
Alright, I've modified the `create_random_world` function to take in a position so that we can have multiple worlds, and made sure to clear the vector upon regeneration.

Reply:
> Right, I kinda rushed rebasing on top of the collision so I didn't notice the overlapping. As for the randomness though, it seems to generate a different world every time so I'm not sure if there's any problem there?

You have to pass a seed otherwise seed 42 is used as default.

Reply:
What's wrong with that?

Reply:
If it is intentional everything is fine.

Reply:
Sorry, I should've explained more. Since the RNG stuff is marked as `static` in the `create_random_world` function, they are globals and so they keep their state throughout the entire program. So the initial seed passed (in this case `42`) is only used the first time `create_random_world` is called.